src/cpp/shared_core/system/PosixSystem.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+ * PosixSystem.cpp
+ *
+ * Copyright (C) 2009-19 by RStudio, Inc.
+ *
+ * Unless you have received this program directly from RStudio pursuant to the terms of a commercial license agreement
+ * with RStudio, then this program is licensed to you under the following terms:
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+ * Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+ * WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+#include <shared_core/system/PosixSystem.hpp>
+
+#include <csignal>
+#include <grp.h>
+#include <memory.h>
+#include <pwd.h>
+#include <sys/prctl.h>
+
+#include <shared_core/Error.hpp>
+#include <shared_core/system/User.hpp>
+
+namespace rstudio {
+namespace core {
+namespace system {
+namespace posix {
+
+Error enableCoreDumps()
+{
+#ifndef __APPLE__
+   int res = ::prctl(PR_SET_DUMPABLE, 1);
+   if (res == -1)
+      return systemError(errno, ERROR_LOCATION);
+#endif
+
+   return Success();
+}
+
+Error ignoreSignal(int in_signal)
+{
+   struct sigaction sa;
+   ::memset(&sa, 0, sizeof(sa));
+
+   sa.sa_handler = SIG_IGN;
+   int result = ::sigaction(in_signal, &sa, nullptr);
+   if (result != 0)
+   {
+      Error error = systemError(result, ERROR_LOCATION);
+      error.addProperty("signal", in_signal);
```

Comment:
Is there any easy way for us to convert the signal number to the signal name here? (just makes debugging a tad easier)

Reply:
In the IDE the `system::ignoreSignal(SignalName name)` function is still available from `System.hpp`. I just need this in the SDK to ignore SIG_PIPE and that's it. I actually had an `ignoreSigPipe` function, but I thought it would be better to not duplicate the code.

Also, I didn't want to move the `SignalName` enum because it's part of `System.hpp` as opposed to `PosixSystem.hpp`, so it may be used on other platforms. I'm open to suggestions on how to resolve that though.

Reply:
It looks like some platforms may have the `strsignal()` function for this purpose: https://linux.die.net/man/3/strsignal